Core/MarketAnalyzer/BaseAnalyzer.cs

```diff
@@ -429,7 +429,29 @@ public static void SaveMarketInfosToFile(Dictionary<string, MarketInfo> marketIn
             if (marketTrendChanges != null && marketTrendChanges.Count > 0)
             {
 
-              double averageTrendChange = marketTrendChanges.Average(mtc => mtc.TrendChange);
+
+            double totalTrendChange = 0;
+
+            foreach (MarketTrendChange marketTrendChange in marketTrendChanges) 
+              {
+                if (marketTrend.IgnoreOutlier != 0) 
+                {
+                  if ((marketTrendChange.TrendChange > marketTrend.IgnoreOutlier) || (marketTrendChange.TrendChange < (marketTrend.IgnoreOutlier * -1)))
+                  {
+                    log.DoLogWarn("Market trend '" + marketTrend.Name + "' is ignoring the outlier '" + marketTrendChange.Market + ".");
+                  }
+                  else
+                  {
+                    totalTrendChange += marketTrendChange.TrendChange;
+                  }
+                }
+                else
+                {
+                    totalTrendChange += marketTrendChange.TrendChange;
+                }
+              }
+
+              double averageTrendChange = totalTrendChange / marketTrendChanges.Count;
```

Comment:
This average calculation is wrong as you should only be dividing against the number of entries that contributed to the total. You need to maintain a count alongside adding to the total and divide by this to find the average.

Reply:
Good catch.  Thanks.

Reply:
This should fix it:  https://github.com/PTMagicians/PTMagic/pull/231/commits/65bb1dc719c6472c0a0033f0b0b2d32d494d3a95